File: backend/model/Refeicao.js
class Refeicao {
    constructor({id = '', nome, alimentos, observacoes}) {
        this.id = id;
        this.nome = nome;
        this.alimentos = alimentos;
        this.observacoes = observacoes;
    }
    toFirestore() {
        return {
            nome: this.nome,
            alimentos: this.alimentos,
            observacoes: this.observacoes
        };
    }

}
module.exports = Refeicao;

File: healthway_app/lib/models/refeicao.dart
class Refeicao {
  final String? id;
  String nome;
  String observacoes;
  Map<String, String> alimentos;

  Refeicao({
    this.id,
    required this.nome,
    required this.observacoes,
    required this.alimentos,
  });

  factory Refeicao.fromJson(Map<String, dynamic> json) {
    return Refeicao(
      id: json['id'] ?? '',
      nome: json['nome'] ?? '',
      observacoes: json['observacoes'] ?? '',
      alimentos: Map<String, String>.from(json['alimentos'] ?? {}),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nome': nome,
      'observacoes': observacoes,
      'alimentos': alimentos,
    };
  }
}
